### PR TITLE
Enable emulator tests

### PR DESCRIPTION
**Pull request by msfjarvis**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description
Enable running instrumentation tests in CI for pull request tests.

## :bulb: Motivation and Context
Since #707 is adding critical instrumentation tests we should start verifying them in CI

## :green_heart: How did you test it?
That's GitHub's job >:]

## :pencil: Checklist

- [ ] I ran `./gradlew spotlessApply` before submitting the PR
- [x] I reviewed submitted code

### CODE REVIEW

**Comment by msfjarvis** [comment]
(I pushed #707 into this branch for testing purposes, I'll merge #707 and rebase this after)

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -3,7 +3,10 @@ on: pull_request
 name: Check pull request
 jobs:
   test-pr:
-    runs-on: ubuntu-latest
+    runs-on: macos-latest
```
Comment (FabianHenneke) [inline .github/workflows/pull_request.yml:6]:
```suggestion
    runs-on: ubuntu-latest
```
I just want to verify whether the incredibly long delay is due this running on macOS.

Comment (msfjarvis) [reply .github/workflows/pull_request.yml]: Ubuntu image does not ship with hardware acceleration so I'm sure it's gonna take much longer.

Comment (FabianHenneke) [reply .github/workflows/pull_request.yml]: Wait, but aren't our remaining tests all unit tests without UI interactions anyway? 

I was also confused by the Actions UI claiming that all four individual jobs had passed, but the meta job was still awaiting its result. Do you know what was up with that?

Comment (msfjarvis) [reply .github/workflows/pull_request.yml]: > Wait, but aren't our remaining tests all unit tests without UI interactions anyway?

That's correct yeah. We can probably even make the new instrumentation tests work in the JVM by mocking `android.net.Uri` behavior and avoid this entire problem altogether.

> I was also confused by the Actions UI claiming that all four individual jobs had passed, but the meta job was still awaiting its result. Do you know what was up with that?

The awaiting result problem is a bug that sometimes affects matrix builds. Waiting a few extra seconds usually resolves it for me.

Comment (FabianHenneke) [reply .github/workflows/pull_request.yml]: > The awaiting result problem is a bug that sometimes affects matrix builds. Waiting a few extra seconds usually resolves it for me.

17 hours in and counting... Well, but Ubuntu broke this for API level 29 anyway, so I will revert this change.

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -4,6 +4,9 @@ name: Check pull request
 jobs:
   test-pr:
     runs-on: ubuntu-latest
```
Comment (FabianHenneke) [inline .github/workflows/pull_request.yml:6]:
```suggestion
    runs-on: macos-latest
```

**Comment by msfjarvis** [comment]
@FabianHenneke you can approve and merge this if you don't want to change anything, it's done from my end.

**Comment by FabianHenneke** [comment]
> @FabianHenneke you can approve and merge this if you don't want to change anything, it's done from my end.

Unfortunately the `test-pr` check is still listed as "Expected", so I can't merge this even though I want to.

**Comment by FabianHenneke** [comment]
Weird, in the GitHub app the tests pass, on the web they don't. Furthermore, I'm not authorized to merge this, even though I could approve it. Count me confused...

**Comment by msfjarvis** [comment]
> Weird, in the GitHub app the tests pass, on the web they don't. Furthermore, I'm not authorized to merge this, even though I could approve it. Count me confused...

Then I suppose GitHub managed to break something during the recent flurry of deployments. Will report to them soon.

**Pull request merged**